### PR TITLE
Send email alerts when Testgrid flows are bad

### LINKED ISSUE #260: Alert when tests in Testgrid start failing
This is specially important for CI flows that aren't master (and thus tend to be forgotten), like release branches.

### PR DESCRIPTION
**Pull request by adrcunha**
Alert emails will be sent:
* if a test fails 3 times in a row
* if a test doesn't run for 24h

Fixes #260.

### CODE REVIEW

File: ci/testgrid/config.yaml [hunk 1]
```diff
@@ -49,6 +49,8 @@ default_dashboard_tab:
   num_columns_recent: 10
   code_search_url_template:      # The URL template to visit when searching for changelists
     url: https://github.com/knative/serving/compare/<start-custom-0>...<end-custom-0>
+  alert_options:
+    alert_mail_to_addresses: 'knative-test+testgrid-alerts@googlegroups.com'
```
Comment (jessiezcc) [inline ci/testgrid/config.yaml]: can we send alert to knative-testinfradev@googlegroups.com? 
I can work with Mark to get such group created, and we will use it for all purpose test infra alerts.

Comment (adrcunha) [reply ci/testgrid/config.yaml]: Done. Holding until the new group is created. I deleted the other group.

/hold

File: ci/testgrid/config.yaml [hunk 2]
```diff
@@ -49,6 +49,8 @@ default_dashboard_tab:
   num_columns_recent: 10
   code_search_url_template:      # The URL template to visit when searching for changelists
     url: https://github.com/knative/serving/compare/<start-custom-0>...<end-custom-0>
+  alert_options:
+    alert_mail_to_addresses: 'knative-testinfradev@googlegroups.com'
```
Comment (jessiezcc) [inline ci/testgrid/config.yaml]: Created knative-productivity-dev@googlegroups.com to use for alerting purpose.

Comment (adrcunha) [reply ci/testgrid/config.yaml]: Updated.

**Pull request merged**